python/lsst/ip/isr/ampOffset.py

```diff
@@ -18,88 +18,412 @@
-    """Calculate and apply amp offset corrections to an exposure.
-    """
+class AmpOffsetTask(Task):
+    """Calculate and apply amp offset corrections to an exposure."""
+
     ConfigClass = AmpOffsetConfig
     _DefaultName = "isrAmpOffset"
 
     def __init__(self, *args, **kwargs):
-        super().__init__(**kwargs)
-        # always load background subtask, even if doBackground=False;
-        # this allows for default plane bit masks to be defined
+        super().__init__(*args, **kwargs)
+        # Always load background subtask, even if doBackground=False;
+        # this allows for default plane bit masks to be defined.
         self.makeSubtask("background")
         if self.config.doDetection:
             self.makeSubtask("detection")
 
     def run(self, exposure):
         """Calculate amp offset values, determine corrective pedestals for each
-        amp, and update the input exposure in-place. This task is currently not
-        implemented, and should be retargeted by a camera specific version.
+        amp, and update the input exposure in-place.
+
+        Parameters
+        ----------
+        exposure: `lsst.afw.image.Exposure`
+            Exposure to be corrected for amp offsets.
+        """
+
+        # Generate an exposure clone to work on and establish the bit mask.
+        exp = exposure.clone()
+        bitMask = exp.mask.getPlaneBitMask(self.background.config.ignoredPixelMask)
+
+        # Fit and subtract background.
+        if self.config.doBackground:
+            maskedImage = exp.getMaskedImage()
+            bg = self.background.fitBackground(maskedImage)
+            bgImage = bg.getImageF(self.background.config.algorithm, self.background.config.undersampleStyle)
+            maskedImage -= bgImage
+
+        # Detect sources and update cloned exposure mask planes in-place.
+        if self.config.doDetection:
+            schema = SourceTable.makeMinimalSchema()
+            table = SourceTable.make(schema)
+            # Detection sigma, used for smoothing and to grow detections, is
+            # normally measured from the PSF of the exposure. As the PSF hasn't
+            # been measured at this stage of processing, sigma is instead
+            # set to an approximate value here (which should be sufficient).
+            _ = self.detection.run(table=table, exposure=exp, sigma=2)
+
+        # Safety check: do any pixels remain for amp offset estimation?
+        if (exp.mask.array & bitMask).all():
+            self.log.warning("All pixels masked: cannot calculate any amp offset corrections.")
+        else:
+            # Set up amp offset inputs.
+            im = exp.image
+            im.array[(exp.mask.array & bitMask) > 0] = np.nan
+            amps = exp.getDetector().getAmplifiers()
+
+            min_dimension = np.min([np.min(amp.getBBox().getDimensions()) for amp in amps])
+            if self.config.ampEdgeWindow is None:
+                self.config.ampEdgeWindow = min_dimension
```

Comment:
Setting values in the config won't work here, as the config has been frozen.  Some other parameter needs to be used to hold this size.

Reply:
Also, if this check is only for the downstream `np.convolve`, then you can push this check/setting there instead.  If the default setting is to set it to the smaller size, then there's no reason to `raise` here.  You should allow smaller sizes in the config, as there may be circumstances where the smaller size is better.

Reply:
Pushed the check to the downstream `np.convolve`. Smaller sizes are allowed in the config.